lib/plugins/aws/deploy/lib/uploadArtifacts.test.js

```diff
@@ -286,33 +286,33 @@ describe('uploadArtifacts', () => {
 
       const uploadZipFileStub = sinon
         .stub(awsDeploy, 'uploadZipFile').resolves();
-      sinon.stub(fs, 'statSync').returns({ size: 1024 });
+      const statSyncStub = sinon.stub(fs, 'statSync').returns({ size: 1024 });
 
       return awsDeploy.uploadFunctions().then(() => {
         expect(uploadZipFileStub.calledTwice).to.be.equal(true);
         expect(uploadZipFileStub.args[0][0])
           .to.be.equal(awsDeploy.serverless.service.functions.first.package.artifact);
         expect(uploadZipFileStub.args[1][0])
           .to.be.equal(awsDeploy.serverless.service.package.artifact);
-        awsDeploy.uploadZipFile.restore();
-        fs.statSync.restore();
+        uploadZipFileStub.restore();
```

Comment:
This is wrong in case id the test fails. You need to move the restores into a `.finally()` at the end of the promise chain to guarantee that they are executed, regardless which branch the promise chain takes. Otherwise the test will pollute the test environment for all tests executed afterwards.